src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -272,7 +275,7 @@ def _add_top_level_record_to_purpose_or_feature_section(
     # Transform the base gvl record into the TCF record type that has more elements for TCF display.
     # Will be a top-level section.
     top_level_tcf_record: NonVendorRecord = tcf_component_type(
-        **fideslang_gvl_record.dict()
+        **fideslang_gvl_record.dict(), retention_period=declaration.retention_period
```

Comment:
If we're not displaying retention periods on the Purpose tab (?), I don't think we should add the retention period here, instead, adding them only to the embedded purposes under vendors, so their retention periods are allowed to vary.

---

This builds a top-level record like a Purpose record, which is also used later as the basis for the EmbeddedPurpose records under Vendors and Systems as well.  If we haven't seen the Purpose thus far, we create a new one, **otherwise we use the existing Purpose**. 

 This hasn't been an issue thus far because the purpose data we're displaying is supposed to be the same everywhere. My concern is that the contents of the Purpose records now can vary.  One System may have a retention period on Purpose 1 of a 100 days. Another System may have a retention period on Purpose 1 of 300 days.  The first data use encountered for a given purpose is the one whose retention period will be used everywhere.  



Even if the GVL enforces that purpose 1 has a retention period of 90 days or similar everywhere across every vendor (does it? I don't know), systems outside of the GVL could define whatever retention period they wish, which could be the value that is used everywhere.

Reply:
yeah you're right! okay so I guess I only want `retention_period` on the embedded purpose. I was having trouble figuring out how to get that data over to the embedded purpose though. let me try again...

Reply:
whoops, i totally missed this - @pattisdr you're right, this needs to go on the `EmbeddedPurpose`, the retention period applies to a given purpose/vendor combination, not globally to a purpose, and there's nothing (that i know of) indicating what the retention period should be for a given purpose generally. this is specified per-vendor, so let's make sure we represent that in our overlay too.

thanks for catching this, i completely glossed over it!

Reply:
We can remove this now since we're no longer adding at the top level: `retention_period=declaration.retention_period`